lib/dialyzer/test/small_SUITE_data/src/guards.erl

```diff
@@ -145,6 +145,14 @@ t17(X) when #{x => X} ->
 t18(X) when <<X>> ->
     ok.
 
+%% Should give a warning, see https://github.com/erlang/otp/issues/7181
+t19() when is_function(ok, -9223372036854775808) ->
+    never.
```

Comment:
There will indeed be a warning:

```
guards.erl:149:12: Guard test is_function
         ('ok',
          -9223372036854775808) can never succeed
```

To ensure that the test case will not fail, you will have to add this warning to [file with expected output](https://github.com/erlang/otp/blob/master/lib/dialyzer/test/small_SUITE_data/results/guards).